api/unstable/async_offload.h

```diff
@@ -0,0 +1,83 @@
+/*
+* Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+*
+* Licensed under the Apache License, Version 2.0 (the "License").
+* You may not use this file except in compliance with the License.
+* A copy of the License is located at
+*
+*  http://aws.amazon.com/apache2.0
+*
+* or in the "license" file accompanying this file. This file is distributed
+* on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+* express or implied. See the License for the specific language governing
+* permissions and limitations under the License.
+*/
+
+#pragma once
+
+#include <s2n.h>
+
+/**
+ * @file async_offload.h
+ * 
+ * The following APIs enable applications to offload expensive handshake operations that do not require user input.
+ * Use s2n_config_set_async_offload_callback() to configure the async callback and the set of operations to offload.
```

Comment:
I wonder whether this is too vague? Since this is unstable and not included in the user guide, should we provide some guidance on when this might be useful, what would count as an expensive handshake operation, and maybe the relationship between this and pkey offload?